crates/sui-storage/src/event_store/sql.rs

```diff
@@ -214,34 +272,43 @@ const QUERY_BY_MODULE: &str = "SELECT * FROM events WHERE timestamp >= ? AND \
 const QUERY_BY_CHECKPOINT: &str =
     "SELECT * FROM events WHERE checkpoint >= ? AND checkpoint <= ? LIMIT ?";
 
-fn check_limit(limit: usize) -> Result<(), EventStoreError> {
+fn check_limit(limit: usize) -> Result<(), SuiError> {
     if limit <= MAX_LIMIT {
         Ok(())
     } else {
-        Err(EventStoreError::LimitTooHigh(limit))
+        Err(SuiError::TooManyItemsError(MAX_LIMIT as u64))
     }
 }
 
 #[async_trait]
 impl EventStore for SqlEventStore {
-    type EventIt = std::vec::IntoIter<StoredEvent>;
-
     async fn add_events(
         &self,
         events: &[EventEnvelope],
         checkpoint_num: u64,
-    ) -> Result<(), EventStoreError> {
+    ) -> Result<(), SuiError> {
+        let mut cur_seq = self.seq_num.load(Ordering::Acquire);
+        let initial_seq = cur_seq;
+
         // TODO: benchmark
         // TODO: use techniques in https://docs.rs/sqlx-core/0.5.13/sqlx_core/query_builder/struct.QueryBuilder.html#method.push_values
         // to execute all inserts in a single statement?
         // TODO: See https://kerkour.com/high-performance-rust-with-sqlite
         for event in events {
+            // Skip events that have a lower sequence number... which must be same or increasing
+            if event.seq_num < cur_seq {
+                debug!(tx_digest =? event.tx_digest, seq_num = event.seq_num, cur_seq, "Skipping event with lower sequence number than current");
+                continue;
```

Comment:
In the way that we currently use `add_events`, the vec of events are from the same txn, so we can directly return here

Reply:
This code is designed for if we ingest more than just one transaction at a time, so would like to keep it for the future